caddyhttp/proxy/proxy.go

```diff
@@ -59,7 +61,10 @@ type UpstreamHost struct {
 	WithoutPathPrefix string
 	ReverseProxy      *ReverseProxy
 	Fails             int32
-	Unhealthy         bool
+	// This is an int32 so that we can use atomic operations to do concurrent
+	// reads & writes to this value.  The default value of 0 indicates that it
+	// is healthy and any non-zero value indicates unhealthy.
+	Unhealthy int32
```

Comment:
Just to confirm, this is int32 so it's not subject to the last bug on this page, right? https://golang.org/pkg/sync/atomic/#pkg-note-BUG - if it is, this would be exposed on ARM systems, for example (Raspberry Pi).